src/main/webapp/app/lecture/pdf-preview/pdf-preview.component.ts

```
@@ -334,8 +400,128 @@ export class PdfPreviewComponent implements OnInit, OnDestroy {
+    }
+
+    /**
+     * Adds a selected PDF file at the end of the current PDF document.
+     * @param event - The event containing the file input.
+     */
+    async mergePDF(event: Event): Promise<void> {
+        const file = (event.target as HTMLInputElement).files?.[0];
+
+        this.isPdfLoading = true;
+        try {
+            const newPdfBytes = await file!.arrayBuffer();
```

Biome

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)